src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/restapi/controller/AdminController.kt

```diff
@@ -39,4 +46,28 @@ class AdminController(
   )
   fun updateByNumbers(@Parameter(required = true) @Valid @RequestBody prisonNumbers: List<String>) =
     personService.updatePeople(prisonNumbers)
+
+  @PutMapping("/referrals/updatePom")
+  @Operation(
+    summary = "Update referrals to update primary and secondary POMs",
+    tags = ["Admin"],
+  )
+  fun updatePoms(): ResponseEntity<String> {
+    referralService.getPrisonIdsWithNoPrimaryPom().forEach {
+      log.info("**** START: Updating POMs for prisoner $it")
+      try {
+        val (primaryPom, secondaryPom) = staffService.getOffenderAllocation(it)
+        referralService.updatePoms(it, primaryPom, secondaryPom)
+        log.info("**** FINISH: Updating POMs for prisoner $it")
+      } catch (ex: Exception) {
+        log.info("**** ERROR: Updating POMs for prisoner $it - ${ex.message}")
```

Comment:
Please can you use the `log.info("message", ex)` form, as we lose the stacktrace otherwise.

Also, If it's a genuine exception, the level should be more appropriate, say warn or error at least.

Reply:
This was initially set to log the full exception, but considering the issues we had seen with POMs i thought to not log the exception and pollute the logs

Reply:
Ok, fine for now. Going forward we should always log the full exception in the form suggested, as it's best practice and will help us investigate any potential issues.